cypress-io/cypress
# fix: Detect CT framework when unconfigured project opened with `--component`

- Closes #22777 

### Additional details
The onboarding wizard automatically detects your CT framework and bundler when advancing from the testing type selection screen, but if you open the launchpad directly into CT using the `--component` flag this logic was being bypassed.  This PR performs the necessary wizard initialization if a testing type is already selected when `open` mode is being set up.

### Steps to test
1. Create a new project without any CT config
    `npm create vite@latest`
    choose React, remaining options can be anything
    `npm i` in project dir to install dependencies.
2. Using current production, open the project: `npx cypress open --component --project <project dir>`
3. Observe that launchpad opens to the "Project setup" screen and no framework is auto-selected
4. Close Cypress, checkout this branch
5. Using this branch, reopen the project: `yarn dev --component --project <project dir>`
6. Observe that launchpad auto-populates the framework and bundler fields with React + Vite

### How has the user experience changed?
**Before**
![Screenshot 2023-03-30 at 8 54 05 AM](https://user-images.githubusercontent.com/11482842/228865258-88dd80bb-8db2-4d31-8ba7-3e82ed2716d0.png)

**After**
![Screenshot 2023-03-30 at 9 12 24 AM](https://user-images.githubusercontent.com/11482842/228865309-651aaaf6-99ba-4164-a1a1-b348778e34dd.png)

### PR Tasks
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
